### DISCUSSION TITLE
How to generate directivity files for other speakers?

### DISCUSSION THREAD

**Discussion by danroth-nyt** [Ifsttar/I-Simpa#367] [Q&A]
Hi, I'm hoping to test a variety of speakers in I-Simpa.  I see some folks have Matlab scripts to produce the directivity files currently included (speaker-test-1 through 3).  How can I generate these for different speakers and then import them into I-Simpa to be recognized as alternative directivity configurations?

**Comment by nicolas-f** [comment]
Hi,

The directivity file is in EASE 7 import format, which is quite simple. Lines starting with the ' ;' character are comments. The header
The header is as follows:

"FileType", "Speaker Types 
"Format",4.0
"LengthUnit", "m


This is followed by a frequency header for each desired one-third octave frequency:

"SpeakerName", "V500 Bass Reflex"
"Manufacturer", "Sample Audio
"Frequency",100,"Hz
"Sensitivity",99.9,"dB"
"Impedance",6,"ohms"
"Q",4.8
"Efficiency",3.9,"%"
"MaxPower",600,"W
"DataGood",1

and frequency measurements:

 " 0°"," 5°"," 10°"," 15°"," 20°"," 25°", ...
 " 0°", 0.00, 0.40, 1.14, 2.26, 3.73, 5.52, ...
 " 5°", 0.00, 0.43, 1.19, 2.32, 3.78, 5.52, ...
 " 10°", 0.00, 0.46, 1.22, 2.35, 3.77, 5.45, ...
 " 15°", 0.00, 0.47, 1.24, 2.35, 3.69, 5.32, ...
 " 20°", 0.00, 0.49, 1.25, 2.32, 3.56, 5.12, ...
 " 25°", 0.00, 0.49, 1.24, 2.27, 3.39, 4.85, ...

Each measurement is associated with its coordinates on a sphere around the source.
